src/functTest/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImageFunctionalTest.groovy

```diff
@@ -137,15 +139,14 @@ class DockerBuildImageFunctionalTest extends AbstractGroovyDslFunctionalTest {
         noExceptionThrown()
     }
 
-    @PendingFeature
     def "task can be up-to-date"() {
         given:
         buildFile << imageCreation()
         buildFile << """
             task verifyImageId {
                 dependsOn buildImage
                 doLast {
-                    assert buildImage.imageId.isPresent()
+                    assert buildImage.imageIdFile.get().asFile.exists()
```

Comment:
Why was this changed exactly? We are testing this already in the assertions as far as I can see: `imageIdFile.isFile()`.

Reply:
imageId is @internal in DockerBuildImage class... I decided to leave the field visibility like you set it and check something else closely related. I am not using the imageId anywhere, but I'm using the imageIdFile so why check imageId? I can change it back to check imageId.isPresent() if you prefer. Or I can remove this assert altogether. It's true we check for the file presence just bellow.

Let me know what you prefer